imp/TokenSet.cs

```diff
@@ -175,6 +175,20 @@ private TokenSet()
             // no operation
         } /* end TokenSet */
 
+        /* ---------------------------------------------------------------------------
+        * private constructor TokenSet (uint[])
+        * ---------------------------------------------------------------------------
+        * used to instantiate the prefabricated first and follow lists
+         * ------------------------------------------------------------------------ */
+
+        public TokenSet(params uint[] TSB)
+        {
+            dataStored.segments[0] = TSB[0];
+            dataStored.segments[1] = TSB[1];
+            dataStored.segments[2] = TSB[2]; 
+            dataStored.elemCount = TSB[3];
+        }
+

```

Comment:
I do not like the name `dataStored`. It is entirely non-descriptive.

If you had no idea what this code is about and all you saw was a single line of code that references the struct like

```
newSet.dataStored.segments[segmentIndex] = 0;
```

does the "dataStored" part in there add any information or does it add confusion?

Either use a descriptive name, like `bits` ...
```
newSet.bits.segments[segmentIndex] = 0;
newSet.bits.elemCount = 0;
```

... or get rid of the struct and use just `segments` and `elemCount` ...

```
newSet.segments[segmentIndex] = 0;
newSet.elemCount = 0;
```

The latter is generally preferable UNLESS we need to assign a compound literal directly to the struct, like so

```
newSet.bits = { 0x0, 0x0, 0x11, 2 };
```

I have not seen your code representing gen_first_sets.c and gen_follow_sets.c. It depends on how you implement those whether you want `segments` and `elemCount` to be within a struct or separate instance variables.